plugins/branch/branch.plugin.zsh

```diff
@@ -0,0 +1,26 @@
+# Branch: displays the current Git or Mercurial branch fast.
+# Victor Torres <vpaivatorres@gmail.com>
+# Oct 2, 2015
+
+function branch_prompt_info() {
+  # Defines path as current directory
+  path=$(pwd)
+  # While current path is not root path
+  while [ $path != '/' ];
+  do
+    # Git repository
+    if [ -d ${path}/.git ];
+    then
+      echo '±' $(/bin/cat ${path}/.git/HEAD | /usr/bin/cut -d / -f 3-)
```

Comment:
You can use `command` before a command to force using the actual command (instead of, say, an alias or a function). So `command cat` and `command cut` will do. 

The `cut` part could also be optimized out with a bit of zsh magic: 
```zsh
echo ${"$(command cat "$var/.git/HEAD")"##*/}
```
and we can optimize out the `cat` part as well with `$(<"$var/.git/HEAD")`:
```zsh
${"$(<"$var/.git/HEAD")"##*/}
```

I know it looks really ugly, but it doesn't run any fork so it should be twice as fast.